### PR TITLE
[#1018] Add a flag to disable shell colors and move verbose logs to d…

### LINKED ISSUE #1018: [statusd] Add a flag to disable shell colors and move verbose logs to debug level
# Problem

When running `statusd` in a docker and not using `-logfile`, each log entry, by default, has shell colors added. 

Also, in `statusd`, we log all sent signals with `INFO` level. These logs can be huge especially if sending `EventDiscoverySummary` signal.

# Acceptance Criteria

1. Provide a flag to disable shell colors from the logger (check out `logutils.OverrideRootLog()` call in `cmd/statusd/main.go`),
2. Change log level in `TriggerDefaultNodeNotificationHandler()` to `DEBUG` or `TRACE`.

### PR DESCRIPTION
**Pull request by adriacidre**
…ebug level

- [x] Added new flag to statusd `disable-log-colors`
- [x] Changed log level in TriggerDefaultNodeNotificationHandler() to TRACE.

Closes #1018

### CODE REVIEW

File: cmd/statusd/main.go [hunk 1]
```diff
@@ -54,6 +54,7 @@ var (
 	pprofPort         = flag.Int("pprofport", 52525, "Port for runtime profiling via pprof")
 	logLevel          = flag.String("log", "INFO", `Log level, one of: "ERROR", "WARN", "INFO", "DEBUG", and "TRACE"`)
 	logFile           = flag.String("logfile", "", "Path to the log file")
+	disableLogColors  = flag.Bool("disable-log-colors", false, "Disables log colors")
```
Comment (adambabik) [inline cmd/statusd/main.go]: To keep it close to other log-related flags, it should start with `log`. For example, `log-without-colors` or `log-enable-colors` with `true` as a default.

Comment (adriacidre) [reply cmd/statusd/main.go]: I'm better going for `log-without-colors` as enabled colors is the default atm

**Comment by adriacidre** [comment]
@adambabik I've pushed a [second commit](https://github.com/status-im/status-go/pull/1020/commits/7cd8e2786459e4a5279fdf9e07b00f9676c225a7) with the code detection for a pseudo-tty, but it introduces some new dependencies, not sure if we want to go for it.

tty detection can be tested like:
```
$ ./build/bin/statusd                # is a terminal
$ echo "boo" | ./build/bin/statusd   # not a terminal
```

File: cmd/statusd/main.go [hunk 2]
```diff
@@ -109,7 +112,12 @@ func main() {
 		return
 	}
 
-	if err := logutils.OverrideRootLog(config.LogEnabled, config.LogLevel, config.LogFile, true); err != nil {
+	colors := !(*logWithoutColors)
+	if colors {
+		colors = terminal.IsTerminal(syscall.Stdin)
```
Comment (adambabik) [inline cmd/statusd/main.go]: `os.Stdin.Fd()` to avoid `syscall`.

**Review by adambabik — APPROVED** [review]
@adriacidre I think it's fine. It's not a huge dependency. I also found this one: https://github.com/mattn/go-isatty Not sure if it's smaller.

LGTM!

**Comment by adambabik** [comment]
@adriacidre also feel free to skip `TestRequestMessagesInGroupChat` if it's flaky.

**Comment by adriacidre** [comment]
@adambabik yep, https://github.com/mattn/go-isatty is much smaller, but wasn't able to make it work :/

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/mattn/go-isatty">mattn/go-isatty</a></strong></div><div>Contribute to go-isatty development by creating an account on GitHub.</div></blockquote>

**Comment by adambabik** [comment]
@adriacidre bummer :/ I guess that's the lib I had issues with as well. We can leave the current one if it's working fine.

File: cmd/statusd/main.go [hunk 3]
```diff
@@ -54,6 +55,7 @@ var (
 	pprofPort         = flag.Int("pprofport", 52525, "Port for runtime profiling via pprof")
 	logLevel          = flag.String("log", "INFO", `Log level, one of: "ERROR", "WARN", "INFO", "DEBUG", and "TRACE"`)
 	logFile           = flag.String("logfile", "", "Path to the log file")
+	logWithoutColors  = flag.Bool("log-without-colors", false, "Disables log colors")
```
Comment (pedropombeiro) [inline cmd/statusd/main.go]: I would put singular `Disable log colors` instead, to keep consistent with other options

**Review by pedropombeiro — APPROVED** [review]
Minor comment, LGTM

**Review by sebastiandelgado — APPROVED** [review]

**Pull request merged**